distributor-node/src/services/networking/query-node/api.ts

```diff
@@ -180,20 +184,48 @@ export class QueryNodeApi {
     )
   }
 
-  public getDistributionBucketsWithObjectsByIds(ids: string[]): Promise<DistirubtionBucketWithObjectsFragment[]> {
-    return this.multipleEntitiesQuery<
-      GetDistributionBucketsWithObjectsByIdsQuery,
-      GetDistributionBucketsWithObjectsByIdsQueryVariables
-    >(GetDistributionBucketsWithObjectsByIds, { ids }, 'distributionBuckets')
+  private async getDataObjectsByBagIds(bagIds: string[]): Promise<MinimalDataObjectFragment[]> {
+    const allBagIds = [...bagIds] // Copy to avoid modifying the original array
+    const fullResult: StorageBagWithObjectsFragment[] = []
+    while (allBagIds.length) {
+      const bagIdsBatch = allBagIds.splice(0, 1000)
+      fullResult.push(
+        ...(await this.multipleEntitiesQuery<
+          GetDataObjectsWithBagsByIdsQuery,
+          GetDataObjectsWithBagsByIdsQueryVariables
+        >(GetDataObjectsWithBagsByIds, { bagIds: bagIdsBatch, limit: bagIdsBatch.length }, 'storageBags'))
+      )
```

Comment:
What do you think of adding a small delay between batches? Like 0.5s? I'd rather have this operation take slightly longer but not put such a heavy load on server